src/TwigComponent/src/ComponentPropertiesExtractor.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\TwigComponent;
+
+use Symfony\UX\TwigComponent\Attribute\ExposeInTemplate;
+use Symfony\UX\TwigComponent\Twig\PropsNode;
+use Twig\Environment;
+
+/**
+ * @author Jean-François Lépine <lepinejeanfrancois@gmail.com>
+ *
+ * @experimental This class is not covered by the BC promise yet
+ */
+final class ComponentPropertiesExtractor
+{
+    public function __construct(
+        private readonly Environment $twig,
+    ) {
+    }
+
+    /**
+     * Returns a list of properties from a Component.
+     *
+     * Warning: We do not recommend using this method at runtime, as it is rather slow.
+     *
+     * @return array<string, array{display: string, name: string, type: string, default: mixed}>
+     */
+    public function getComponentProperties(ComponentMetadata $metadata): array
+    {
+        if ($metadata->isAnonymous()) {
+            return $this->getAnonymousComponentProperties($metadata);
+        }
+
+        return $this->getNonAnonymousComponentProperties($metadata);
+    }
+
+    /**
+     * @return array<string, array{display: string, name: string, type: string, default: mixed}>
```

Comment:
```suggestion
     * @return array<string, array{display: string, name: string, type: string, default: ?string}>
```

(once fixed the value json encoding)